src/twisted/logger/_format.py

```diff
@@ -287,3 +287,33 @@ def formatWithCall(formatString, mapping):
     return unicode(
         aFormatter.vformat(formatString, (), CallMapping(mapping))
     )
+
+
+
+def formatEventWithTraceback(event):
+    """
+    Format an event as a unicode string and append a traceback if the event
+    contains a C{event["log_failure"]} key.
+
+    This implementation will always attempt to append a traceback, even if
+    formatting the C{event["log_format"]} fails.  The traceback, if bytes, will
+    be assumed to be UTF-8.
+
+    @param event: A logging event.
+    @type event: L{dict}
+
+    @return: A formatted string with a traceback if applicable.
+    @rtype: L{unicode}
+    """
+    eventText = formatEvent(event)
+    try:
+        traceback = event["log_failure"].getTraceback()
+        if isinstance(traceback, bytes):
+            traceback = traceback.decode('utf-8')
+    except KeyError:
+        return eventText
+    except:
```

Comment:
This should be `except Exception:` - we don't want to catch `SystemExit`!

Reply:
No problem.  I modeled formatEventWithTraceback after formatEventAsClassicLogText (above it), which also uses all caps and an bare except.  Is that something that should be fixed (in a different ticket/PR)?

Reply:
Yes, I think `formatEventAsClassicLogText` should get changed via a new ticket.  I appreciate that you started with a goal of consistency, as well!